src/main/java/cn/nukkit/nbt/NBTIO.java

```diff
@@ -81,7 +81,8 @@ public static Item getItemHelper(CompoundTag tag) {
             BlockState blockState = getBlockStateHelper(block);
             if (blockState != null) item.setBlockUnsafe(blockState.toBlock());
         }
-        if (item.getDamage() != 0) {
+
+        if (item.getDamage() == 0 && damage != 0) {
```

Comment:
Why only update damage to 0?
 It should be
`if (item.getDamage()!=0 && damage !=0)`

Reply:
If you set damage when it is not equal to 0 the damage saved in nbt will not be set when the initial damage of the item is 0, for example potions initially have damage 0, because of this if you throw it away it will turn into a bottle of water. I tested this

`if (item.getDamage() !=0 && damage != 0) {`:
https://youtu.be/W-rxPchFxzo

`if (item.getDamage() == 0 && damage != 0) {`:
https://youtu.be/9g4aq0nWrUw

Reply:
ok why not set the damage directly
`if damage!=0` set